src/fetchers/stats-fetcher.js

```diff
@@ -99,6 +126,34 @@ const totalCommitsFetcher = async (username) => {
   return 0;
 };
 
+const totalStarsFetcher = async (username, repoToHide) => {
+  let nodes = [];
+  let hasNextPage = true
+  let endCursor = null
+  while (hasNextPage) {
+    const variables = { login: username, first: 100, after: endCursor };
+    let res = await retryer(repositoriesFetcher, variables);
+
+    if (res.data.errors) {
+      logger.error(res.data.errors);
+      throw new CustomError(
+        res.data.errors[0].message || "Could not fetch user",
+        CustomError.USER_NOT_FOUND,
+      );
+    }
+
+    const allNodes = res.data.data.user.repositories.nodes;
+    const nodesWithStars = allNodes.filter((node) => node.stargazers.totalCount !== 0)
+    nodes.push(...nodesWithStars);
```

Comment:
Very well designed! Doing this in a separate function will safe us GraphQL points (see https://docs.github.com/en/graphql/overview/resource-limitations). There is one improvement I can still think of. Maybe we can stop calling the GraphQL API if the last repository returned from the last batch has `0` stars. Since we use `orderBy: {direction: DESC, field: STARGAZERS}`, the next pages do not contain repositories with any stars.

Reply:
It's already done. the fetches stops as soon as there is a repo with 0 stars. https://github.com/anuraghazra/github-readme-stats/pull/2100/files#diff-1683fbb5a86a8ceef648d82b913f38fb1f4cca7474cbbb75b7ca2c771f9a2a90R148

There is also a test for it [test](https://github.com/anuraghazra/github-readme-stats/pull/2100/files#diff-aed474a9db52cb097696aa8b70c6207f3fccd10a76a8e0beca7d1b82ac8caf0cR132)

Reply:
Your right! Sorry, I overlooked line 148. Great 😄.